### PR TITLE
Validate public address value [HZ-1738]

### LINKED ISSUE #21584: Member accepts blank public address
**Describe the bug**
Start a member with "-Dhazelcast.local.publicAddress=" does not fail fast.
Gives
```
Members {size:1, ver:1} [
	Member []:5701 - 33b52e83-504e-42ee-9799-fa66f651e489 this
]
```

**Expected behavior**
Throw exception

**To Reproduce**
As above, `java -Dhazelcast.local.publicAddress= -jar ...`

**Additional context**
Tested on `Hazelcast Platform 5.2-SNAPSHOT (20220608 - b03c176)`

### PR DESCRIPTION
**Pull request by vbekiaris**
when blank, fail fast with `IllegalArgumementException`

Fixes #21584 

Breaking changes (list specific methods/types/messages):
* introduces config validation for public address that may break existing invalid configurations

Checklist:
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java [hunk 1]
```diff
@@ -295,6 +295,15 @@ public void testPublicAddress_withSpecifiedHostAndPortViaProperty() throws Excep
         assertEquals(new Address(host, port), addressPicker.getPublicAddress(null));
     }
 
+
+    @Test(expected = IllegalArgumentException.class)
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java:299]: It's so old-fashioned, to use the `expected` param in the `@Test` annotation. :stuck_out_tongue: 
Have you considered the more focused JUnit's `assertThrows` or AssertJ's `assertThatThrownBy`?

File: hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java [hunk 2]
```diff
@@ -303,6 +312,14 @@ public void testPublicAddress_withInvalidAddress() throws Exception {
         addressPicker.pickAddress();
     }
 
+    @Test(expected = IllegalArgumentException.class)
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java:315]: dtto

**Review by kwart — APPROVED** [review]
A little bit of nit-picking.

File: hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java (hunk 1 above)
Comment (vbekiaris) [reply hazelcast/src/test/java/com/hazelcast/instance/impl/DefaultAddressPickerTest.java]: updated in https://github.com/hazelcast/hazelcast/pull/24729/commits/b92e5a8c7a0150ce8504b5806fb47220bd6de359

**Comment by vbekiaris** [comment]
run-lab-run

**Pull request merged**